pommesinvest/model_funcs/subroutines.py

```diff
@@ -1703,3 +1703,131 @@ def create_new_built_storages_myopic_horizon(
+        with component names as keys
+
+    im : :class:`InvestmentModel`
+        The investment model that is considered
+
+    node_dict : :obj:`dict` of :class:`nodes <oemof.network.Node>`
+        Dictionary containing all nodes of the EnergySystem
+
+    Returns
+    -------
+    node_dict : :obj:`dict` of :class:`nodes <oemof.network.Node>`
+        Modified dictionary containing all nodes of the EnergySystem
+        including the electric vehicle elements
+    """
+    # Create ev bus first since this is required for other components
+    for i, c in input_data["electric_vehicles"].iterrows():
+        if c["type"] == "bus":
+            node_dict[i] = solph.buses.Bus(label=i)
+
+    for i, c in input_data["electric_vehicles"].iterrows():
+        component_type = c["type"]
+        if component_type == "transformer":
+            outflow_args = {"nominal_value": c["nominal_value"]}
+            if "_cc" in i:
+                outflow_args["max"] = (
+                    input_data["electric_vehicles_ts"]
+                    .loc[im.start_time : im.end_time, "cc_avail"]
+                    .to_numpy()
+                )
+            elif "_uc" in i:
+                outflow_args["fix"] = (
+                    input_data["electric_vehicles_ts"]
+                    .loc[im.start_time : im.end_time, "uc_avail"]
+                    .to_numpy()
+                )
+            node_dict[i] = solph.components.Transformer(
+                label=i,
+                inputs={node_dict[c["from"]]: solph.flows.Flow()},
+                outputs={node_dict[c["to"]]: solph.flows.Flow(**outflow_args)},
+                conversion_factors={node_dict[c["to"]]: c["efficiency_el"]},
+            )
+        elif component_type == "bus":
+            pass  # has already been created
+        elif component_type == "storage":
+            node_dict[i] = solph.components.GenericStorage(
+                label=i,
+                inputs={node_dict[c["from"]]: solph.flows.Flow()},
+                outputs={node_dict[c["to"]]: solph.flows.Flow()},
+                nominal_storage_capacity=c["nominal_value"],
+                # storage level indexed in TIMEPOINTS
+                max_storage_level=(
+                    input_data["electric_vehicles_ts"]
+                    .loc[
+                        im.start_time : f"{int(im.end_time[:4])+1}"
+                        f"-01-01 00:00:00",
+                        "soc_upper",
+                    ]
+                    .to_numpy()
+                ),
+                min_storage_level=(
+                    input_data["electric_vehicles_ts"]
+                    .loc[
+                        im.start_time : f"{int(im.end_time[:4])+1}"
+                        f"-01-01 00:00:00",
+                        "soc_lower",
+                    ]
+                    .to_numpy()
+                ),
+                inflow_conversion_factor=1,
+                outflow_conversion_factor=1,
```

Comment:
This only works with `solph_sequence(1)`  `from oemof.solph._plumbing import sequence as solph_sequence` in my implementation

Reply:
Interesting, thank you. I assume it is due to your amemendments and the presence of periods in this case, right? In other soloh versions, it should be implicitly casted to a `solph_sequence` with a default of 1, right?